### ISSUE TITLE
is 3.0.0 actually supported?

### ISSUE THREAD

**Issue by progenysw** [SamEdwardes/spaCyTextBlob#7]
here's the error from the space installer
spacytextblob 0.1.7 requires spacy<3.0.0,>=2.3.2, but you have spacy 3.0.0 which is incompatible.

**Comment by SamEdwardes** [comment]
3.0 is not supported yet. It will be soon I am working on an update (https://github.com/SamEdwardes/spaCyTextBlob/tree/spacy3). 

Dependences are defined in [pyproject.toml](https://github.com/SamEdwardes/spaCyTextBlob/blob/main/pyproject.toml) file.

https://github.com/SamEdwardes/spaCyTextBlob/blob/18dfa65e58c721e961487580492a663d5ae43e1b/pyproject.toml#L12-L15

`spacy = "^2.3.2"` means >=2.3.2<3.0.0 [poetry docs](https://python-poetry.org/docs/versions/#caret-requirements).

Is there anywhere else I may have missed? Thank you for raising the issue and your help.

**Comment by plaguss** [comment]
Hi @SamEdwardes, there is another error when adding the `spacy_text_blob` to a pipe on the new 3.0.
```python
>>> import spacy
>>> from spacytextblob.spacytextblob import SpacyTextBlob

>>> nlp = spacy.load('en_core_web_sm')
>>> spacy_text_blob = SpacyTextBlob()
>>> nlp.add_pipe(spacy_text_blob)
```
Fails with the following error:
```
Traceback (most recent call last):
  File "<input>", line 6, in <module>
  File "C:\Users\agustin\git_repository\oiet\venv\lib\site-packages\spacy\language.py", line 748, in add_pipe
    raise ValueError(err)
ValueError: [E966] `nlp.add_pipe` now takes the string name of the registered component factory, not a callable component. Expected string, but got <spacytextblob.spacytextblob.SpacyTextBlob object at 0x00000134979D3490> (name: 'None').
- If you created your component with `nlp.create_pipe('name')`: remove nlp.create_pipe and call `nlp.add_pipe('name')` instead.
- If you passed in a component like `TextCategorizer()`: call `nlp.add_pipe` with the string name instead, e.g. `nlp.add_pipe('textcat')`.
- If you're using a custom component: Add the decorator `@Language.component` (for function components) or `@Language.factory` (for class components / factories) to your custom component and assign it a name, e.g. `@Language.component('your_name')`. You can then run `nlp.add_pipe('your_name')` to add it to the pipeline.
```

This problem is treated in [migrating add_pipe](https://spacy.io/usage/v3#migrating-add-pipe).

This solves the problem (or at least the example from the documentation...)

```python
>>> import spacy
>>> from spacytextblob.spacytextblob import SpacyTextBlob
>>> from spacy.language import Language

>>> @Language.component("text_blob_sentiment")
... def textblob(doc):
...     spacy_text_blob = SpacyTextBlob()
...     return spacy_text_blob(doc)

>>> nlp = spacy.load('en_core_web_sm')
>>> nlp.add_pipe('text_blob_sentiment')
```
Now, in spacy==3.0.0, when printing the pipe names:
```python
>>> print(nlp.pipe_names)
['tok2vec', 'tagger', 'parser', 'ner', 'attribute_ruler', 'lemmatizer', 'text_blob_sentiment']
```
And following with the original example from the documentation:
```python
>>> text = 'I had a really horrible day. It was the worst day ever! But every now and then I have a really good day that makes me happy.'
>>> doc = nlp(text)
>>> print('Polarity:', doc._.sentiment.polarity) 
Polarity: -0.125
>>> print('Sujectivity:', doc._.sentiment.subjectivity) 
Sujectivity: 0.9
```

**Comment by SamEdwardes** [comment]
Closed by #8.
